extensions/amp-next-page/0.1/next-page-service.js

```diff
@@ -457,24 +474,31 @@ export class NextPageService {
       return;
     }
 
-    let documentRef;
+    let documentIndex = i;
     let analyticsEvent = '';
 
-    if (position.relativePos === 'top') {
-      documentRef = this.documentRefs_[i + 1];
-      analyticsEvent = 'amp-next-page-scroll';
-    } else if (position.relativePos === 'bottom') {
-      documentRef = this.documentRefs_[i];
-      analyticsEvent = 'amp-next-page-scroll-back';
+    switch (position.relativePos) {
+      case 'top':
+        documentIndex = i + 1;
+        analyticsEvent = 'amp-next-page-scroll';
+        break;
+      case 'bottom':
+        analyticsEvent = 'amp-next-page-scroll-back';
+        break;
+      default:
+        break;
     }
 
-    if (documentRef && documentRef.amp) {
+    if (
+      this.documentRefs_[documentIndex] &&
```

Comment:
Perhaps we should pull out `this.documentRefs_[documentIndex]` into a variable, instead of re-accessing it each time. Could also just avoid dealing with index in `setActiveDocument_` and just deal with docRefs. The only change would be:

```javascript
// Prevent updating visibility of the host document
if (documentRef === this.documentRefs_[0]) {
  return;
}
```